metricbeat/module/kibana/stats/data_xpack.go

```diff
@@ -0,0 +1,103 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package stats
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/elastic/beats/libbeat/common"
+	s "github.com/elastic/beats/libbeat/common/schema"
+	c "github.com/elastic/beats/libbeat/common/schema/mapstriface"
+	"github.com/elastic/beats/metricbeat/helper"
+	"github.com/elastic/beats/metricbeat/mb"
+)
+
+var (
+	schemaXPackMonitoring = s.Schema{
+		"concurrent_connections": c.Int("concurrent_connections"),
+		"os": c.Dict("os", s.Schema{
+			"load": c.Dict("cpu.load_average", s.Schema{
```

Comment:
@ruflin @jsoriano The unit test for this file is failing because of the usage of `cpu.load_average` here. The failure  message says:

```
1 error: key `os.cpu.load_average` not found
```

 I think it doesn't like the fact that they key is nested (`cpu.load_average`).

Any ideas how I could achieve what I'm trying to over here? I want the resulting schema to look like this:

```json
"os": {
  "load": {
    "1m": 12345,
    "5m": 23456,
    "15m": 34567
  }
}
```

But the corresponding source data looks like this:

```json
"os": {
  "cpu": {
    "load_average": {
      "1m": 12345,
      "5m": 23456,
      "15m": 34567,
    }
  }
}
```

Reply:
@ycombinator try to use the nested notation in `c.Float()`, something like this:
```
"load": s.Object{
  "1m": c.Float("cpu.load_average.1m"),
  ... 
}
```

Reply:
Thanks @jsoriano. I tried that but it didn't work either:

```
=== RUN   TestEventMappingXPack
--- FAIL: TestEventMappingXPack (0.00s)
        Error Trace:    data_xpack_test.go:43
        Error:          Received unexpected error:
                        3 errors: key `os.cpu.load_average.1m` not found; key `os.cpu.load_average.5m` not found; key `os.cpu.load_average.15m` not found
        Test:           TestEventMappingXPack
        Messages:       _meta/test/stats.700.json
        Error Trace:    data_xpack_test.go:44
        Error:          Should be true
        Test:           TestEventMappingXPack
        Messages:       _meta/test/stats.700.json
        Error Trace:    data_xpack_test.go:45
        Error:          Not equal:
                        expected: 0
                        actual  : 1
        Test:           TestEventMappingXPack
        Messages:       _meta/test/stats.700.json
```

Reply:
I have been taking a look to the behaviour of nested fields in schema and I have opened a PR for discussion (#7583). 

The thing is that for some types (`string` and `interface{}`) the map is converted to `common.MapStr`, and then the value is obtained with `GetValue()`, what works with nested field paths. For the rest of types, as dicts and floats the map is used as a normal map and then it cannot work with nested values.  I'm not sure if this is the intended behaviour... but we should probably unify behaviours.